SD_Card_Driver/Core/Inc/LOS_D_sd.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef LOS_D_SD_H_
+#define LOS_D_SD_H_
+
+#include "fatfs.h"
+#include "string.h"
+#include "LOS_D_storage_device.hpp"
+
+class SDCard : StorageDevice {
+	private:
+		FATFS fs;
+		const char* filePath;
+		bool fileOpened = false;
+		FIL file;

```

Comment:
So where are these private member variables being initialized?

Reply:
They're currently initialized in `cleanup`, but as per Aidan's suggestion I'll initialize them in the constructor.